### PR TITLE
Issue #178 Document use of Persistent config options and Env variables

### PR DESCRIPTION
**Pull request by Preeticp**
This PR covers the documentation for controlling runtime behavior of Minishift. As part of this PR: 
- Created a parent section: 'Controlling runtime behavior of Minishift'
- Created a section for Flags to cover all the options to control Minishift behavior 
- Edited the documentation for the Persistent config and env variables section.
- Made a few minor changes wrt formatting etc.

### CODE REVIEW

**Comment by hferentschik** [comment]
@Preeticp could you please rebase this pull request. You somehow squashed your changes into @LalatenduMohanty commit.

File: hack/boilerplate/boilerplate.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ def get_regexs():
     # Search for "YEAR" which exists in the boilerplate, but shouldn't in the real thing
```
Comment (hferentschik) [inline hack/boilerplate/boilerplate.py]: This file does not belong onto this commit

File: hack/boilerplate/boilerplate.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def get_regexs():
...
     regexs["year"] = re.compile( 'YEAR' )
     # dates can be 2014, 2015 or 2016, company holder names can be anything
```
Comment (hferentschik) [inline hack/boilerplate/boilerplate.py]: not this one either ;-)

File: docs/using.md [hunk 3]
```diff
@@ -9,11 +9,12 @@ overview of different components and services.
   - [Starting OpenShift](#starting-openshift)
   - [Stopping OpenShift](#stopping-openshift)
   - [Deleting OpenShift](#deleting-openshift)
-- [Environment variables](#environment-variables)
-- [Persistent configuration](#persistent-configuration)
-  - [Configuration options precedence](#configuration-options-precedence)
-  - [Setting persistent configuration values](#setting-persistent-configuration-values)
-  - [Unsetting persistent configuration values](#unsetting-persistent-configuration-values)
+- [Controlling runtime behavior of Minishift](controlling-runtime-behavior-of-minishift)
```
Comment (hferentschik) [inline docs/using.md]: Maybe "Controlling runtime options" is better, not sure?

File: docs/using.md [hunk 4]
```diff
@@ -34,67 +35,76 @@ This section contains information about basic virtual machine and OpenShift mana
...
 ### Deleting OpenShift
 
-The [minishift delete](./minishift_delete.md) command is used to delete the OpenShift instance.
+The [`minishift delete`](./minishift_delete.md) command is used to delete the OpenShift instance.
 This command shuts down and deletes the Minishift virtual machine. No data or state is preserved.
 
+<a name="controlling-runtime-behavior-of-minishift"></a>
+## Controlling runtime behavior of Minishift  
+
+The runtime behavior of Minishift can be controlled through flags, environment variables, and persistent configuration options, as discussed in the following sections.
```
Comment (hferentschik) [inline docs/using.md]: :+1:, great change overall. I like it.

File: docs/using.md [hunk 5]
```diff
@@ -34,67 +35,76 @@ This section contains information about basic virtual machine and OpenShift mana
...
+
+The following precedence order is applied to control the behavior of Minishift. Each item in the following list takes precedence over
+the item below it:
+
+1. Use a command line flag as specified in the [Flags](flags) section.
+1. Set environment variable as described in the [Environment variables](#environment-variables) section.
+1. Use persistent configuration option as described in the [Persistent configuration](#persistent-configuration) section.
+1. Set the default value as defined by Minishift.
```
Comment (hferentschik) [inline docs/using.md]: You don't 'set' a default value. It gets applied. So just "default value as defined by Minishift". Obviously that breaks the flow of this enumeration, so if you want this to be active, maybe :"Accept the default value as defined by Minishift."

**Comment by Preeticp** [comment]
Ah will do @hferentschik

File: docs/using.md [hunk 6]
```diff
@@ -34,67 +35,76 @@ This section contains information about basic virtual machine and OpenShift mana
...
+
+
+<a name="flags"></a>
+### Flags
+
+You can use command line flags with Minishift to specify options and direct its behavior. This has the highest precedence. Some of the commonly used command line flags are `cpus`, `memory` or `vm-driver`, for a full set of supported flags refer to the Synopsis section of the
+[`minishift config`](./minishift_config.md) sub-command.
```
Comment (hferentschik) [inline docs/using.md]: "for a full set of supported flags refer to the Synopsis section of the
 +[`minishift config`](./minishift_config.md) sub-command.", I skip this part. It is not quite right. There are ways more flags and different commands take different flags. Not all flags can be controlled via persistent configuration though. That are just the ones mentioned in `minishift config`. Environment variables can, however, be used to replace any option of any command. Maybe we can make this distinction clear as well.

Comment (hferentschik) [inline docs/using.md]: "Some of the commonly used command line flags are `cpus`, `memory` or `vm-driver`" => "Some of the commonly used command line flags of the start command are for example `cpus`, `memory` and`vm-driver"`"

Flags are not just for the start command, pretty much all commands have flags/options.

File: docs/using.md [hunk 7]
```diff
@@ -34,67 +35,76 @@ This section contains information about basic virtual machine and OpenShift mana
...
-
-Minishift also maintains a configuration file (`$MINISHIFT_HOME/config/config.json`) which can be
-used to set commonly used command-line flags persistently. For example `cpus`, `memory` or `vm-driver`.
-For a full set of supported configuration options refer to the synopsis of the
-[config](./minishift_config.md) sub-command.
+### Persistent configuration
 
-<a name="configuration-options-precedence"></a>
-### Configuration options precedence
+Using persistent configuration allows you to control Minishift's behavior without specifying actual command line flags, similar to the way you use [environment variables](#environment-variables).
```
Comment (hferentschik) [inline docs/using.md]: This is probably the point to make the distinction that env variables can be used for any odd options, whereas persistent configuration can only be applied to the options listed in config command synopsis (see comment above)

**Comment by hferentschik** [comment]
Add to whitelist

File: docs/using.md (hunk 3 above)
Comment (Preeticp) [inline docs/using.md]: Controlling runtime options, implies that we are controlling the options. So, are we controlling Minishft's behavior through these options or are we controlling the options themselves? AFAIU we customize the options in order to control Minshift's runtime behavior, no?
How about 'Minishift runtime options' ?

**Comment by hferentschik** [comment]
@Preeticp could you please rebase this on top of master. There have been changes to _using.md_ in the meantime and there is a merge conflict around proxies.

File: docs/using.md (hunk 3 above)
Comment (hferentschik) [inline docs/using.md]: +1, I'll can take care of this.

File: docs/using.md [hunk 8]
```diff
@@ -78,57 +79,66 @@ cross origin requests from all domains, one can execute:
...
-Per default the master configuration is targeted, but you can also path the node config
+Per default, the master configuration is targeted, but you can also path the node config
 by specifying the `target` flag. For more details about the
 `set` command refer to its [synopsis](./minishift_openshift_config_set.md).
 
 **Note:** OpenShift will be restarted after applying the patch.
 
+<a name="controlling-runtime-behavior-of-minishift"></a>
+## Controlling runtime behavior of Minishift
+
```
Comment (hferentschik) [inline docs/using.md]: 👍 Looks all very nice, thanks

File: docs/using.md [hunk 9]
```diff
@@ -238,13 +249,18 @@ $ minishift start --http-proxy http://YOURPROXY:PORT --https-proxy https://YOURP
...
 
-There is also `--no-proxy` which allows you to specify a comma-separated list of hosts which
+You can also use `--no-proxy` to specify a comma-separated list of hosts which
 should not be proxied. For a list of all available options refer to the
 [synopsis](./minishift_start.md) of the `start` command.
 
 Using the proxy options will transparently configure the Docker daemon as well as OpenShift to
 use the specified proxies.
 
+It will also set the `http-proxy`, `https-proxy` and `no-proxy`
```
Comment (hferentschik) [inline docs/using.md]: Careful here. You resolved the merge conflict the wrong way. You brought this sentence back, even though it was removed on master. I'll take care of this.

**Comment by hferentschik** [comment]
Merged, thanks.

File: docs/using.md (hunk 9 above)
Comment (Preeticp) [inline docs/using.md]: Ah, sorry about that. Will be careful in the future.

**Pull request closed without merging**